File: CS/WindowsApplication7/PopupMenuShowingEventArgs.cs
using System;
using DevExpress.Utils.Menu;
using DevExpress.XtraTab;

namespace WindowsApplication7 {
    public delegate void PopupMenuShowingEventHandler(object sender, PopupMenuShowingEventArgs e);

    public class PopupMenuShowingEventArgs : EventArgs {
        IXtraTabPage tabPageCore;
        DXPopupMenu menuCore;

        public PopupMenuShowingEventArgs(IXtraTabPage tabPage) {
            tabPageCore = tabPage;
        }
        public DXPopupMenu Menu {
            get { return menuCore; }
            set { menuCore = value; }
        }
        public IXtraTabPage TabPage {
            get { return tabPageCore; }
        }
    }
}

File: VB/WindowsApplication7/PopupMenuShowingEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraTab

Namespace WindowsApplication7
	Public Delegate Sub PopupMenuShowingEventHandler(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs)

	Public Class PopupMenuShowingEventArgs
		Inherits EventArgs
		Private tabPageCore As IXtraTabPage
		Private menuCore As DXPopupMenu

		Public Sub New(ByVal tabPage As IXtraTabPage)
			tabPageCore = tabPage
		End Sub
		Public Property Menu() As DXPopupMenu
			Get
				Return menuCore
			End Get
			Set(ByVal value As DXPopupMenu)
				menuCore = value
			End Set
		End Property
		Public ReadOnly Property TabPage() As IXtraTabPage
			Get
				Return tabPageCore
			End Get
		End Property
	End Class
End Namespace
